DemoApp/StreamChat/StreamChatWrapper.swift

```diff
@@ -0,0 +1,155 @@
+        Appearance.default.localizationProvider = { key, table in
+            let localizedString = localizationProvider(key, table)
+
+            return localizedString == key
+                ? Bundle.main.localizedString(forKey: key, value: nil, table: table)
+                : localizedString
+        }
+    }
+
+    func configureUI() {
+        // Customize UI configuration
+        Components.default.messageListDateSeparatorEnabled = true
+        Components.default.messageListDateOverlayEnabled = true
+
+        // Customize UI components
+        Components.default.channelListRouter = DemoChatChannelListRouter.self
+        Components.default.channelVC = DemoChatChannelVC.self
+        Components.default.messageContentView = DemoChatMessageContentView.self
+        Components.default.messageActionsVC = DemoChatMessageActionsVC.self
+        Components.default.reactionsSorting = { $0.type.position < $1.type.position }
+    }
+}
+
+// MARK: User Authentication
+
+extension StreamChatWrapper {
+    func connect(user: DemoUserType, completion: @escaping (Error?) -> Void) {
+        switch user {
+        case let .credentials(userCredentials):
+            client.connectUser(
+                userInfo: userCredentials.userInfo,
+                token: userCredentials.token,
+                completion: completion
+            )
+        case let .guest(userId):
+            client.connectGuestUser(userInfo: .init(id: userId), completion: completion)
+        case .anonymous:
+            client.connectAnonymousUser(completion: completion)
+        }
+    }
+
+    func logIn(as user: DemoUserType) {
+        // Setup Stream Chat
+        setUpChat()
+
+        // Connect to chat
+        connect(user: user) { [weak self] in
+            if let error = $0 {
+                log.warning(error.localizedDescription)
+            } else {
+                self?.setupRemoteNotifications()
+            }
+        }
+    }
+
+    func logOut() {
+        let currentUserController = client.currentUserController()
+        if let deviceId = currentUserController.currentUser?.currentDevice?.id {
+            currentUserController.removeDevice(id: deviceId) { error in
+                if let error = error {
+                    log.warning("removing the device failed with an error \(error)")
+                }
+            }
+        }
+
+        client.disconnect()
+        client = nil
+    }
+}
+
+// MARK: Controllers
+
+extension StreamChatWrapper {
+    func channelController(for channelId: ChannelId?) -> ChatChannelController? {
+        channelId.map { client.channelController(for: $0) }
+    }
+
+    func channelListController(query: ChannelListQuery) -> ChatChannelListController {
+        client.channelListController(query: query)
+    }
+}
+
+// MARK: Push Notifications
+
+extension StreamChatWrapper {
+    func setupRemoteNotifications() {
+        PushNotifications.shared.registerForPushNotifications()
```

Comment:
what if we pass `PushNotifications` instance as a dependency via `init`?

Reply:
Thanks. I provided closure to fire this logic to `StreamChatWrapper` to which the outer clients can hook into.